optuna/samplers/_tpe/multi_objective_sampler.py

```diff
@@ -557,9 +557,10 @@ def _calculate_weights_below(
                     - self._compute_hypervolume(
                         np.asarray(lvals_below[:i] + lvals_below[i + 1 :]), reference_point
                     )
-                    for i in range(len(lvals))
+                    for i in range(n_below)
```

Comment:
Thank you for fixing the length of `contributions`.
I'm wondering why we need to pad last `len(lvals) - n_below` contributions with the value of `contributions[n_below-1]`.
Was it a simple typo?

Reply:
`calculate_weights_below` should return weights of which length is equal to the number of below observations.
Actually, it returns `np.asarray([])` when `n_below == 0` and `np.asarray([1.0])` when `n_below == 1`.
Therefore, I think that `len(lvals)` is a simple typo for `len(lvals_below)`, that is, `n_below`.

Reply:
`_calculate_weights_below` should return weights of which length is equal to the number of below observations.
Actually, it returns `np.asarray([])` when `n_below == 0` and `return np.asarray([1.0])` when `n_below == 1`.
Therefore, I think that `len(lvals)` is a simple typo for `len(lvals_below)`, that is, n_below.
The reason why we have not noticed this bug is that `zip(weights_below, below)` only loops `min(len(weights_below), len(below)) so that no runtime error occurs.
https://github.com/optuna/optuna/blob/391f2802a589b308821498021f81115a87ad9b4f/optuna/samplers/_tpe/multi_objective_sampler.py#L297

Reply:
Sounds reasonable to fix `len(lvals)` to `n_below`.

Reply:
Thank you for your explanation. That makes sense.